File: client/java/AI.java
import com.sun.jna.Pointer;
import java.util.Random;


///The class implementing gameplay logic.
class AI extends BaseAI
{
  private Random r;

  public String username()
  {
    return "Shell AI";
  }
  public String password()
  {
    return "password";
  }

  //This function is called each time it is your turn
  //Return true to end your turn, return false to ask the server for updated information
  public boolean run()
  {
    String[] s = {"l", "r", "u", "d"};
    for(int i = 0; i < bots.length; i++)
    {
      if(bots[i].getOwner() == playerID())
      {
        bots[i].move(s[r.nextInt(4)]);
        if(bots[i].getBuildRate() != 0)
        {
          bots[i].build(types[r.nextInt(types.length)], bots[i].getX()+1, bots[i].getY(), 1);
        }
      }
    }
    return true;
  }


  //This function is called once, before your first turn
  public void init()
  {
    r = new Random();
  }

  //This function is called once, after your last turn
  public void end() {}
  
  
  public AI(Pointer c)
  {
    super(c);
  }
}

File: client/c/AI.cpp
#include "AI.h"
#include "util.h"

#include <cstdlib>
#include <ctime>

AI::AI(Connection* conn) : BaseAI(conn) {}

const char* AI::username()
{
  return "Shell AI";
}

const char* AI::password()
{
  return "password";
}

string direction[] = {"l","r","u","d"};

//This function is run once, before your first turn.
void AI::init()
{
  srand ( time(NULL) );
}

//This function is called each time it is your turn.
//Return true to end your turn, return false to ask the server for updated information.
bool AI::run()
{
  for(int i = 0; i < bots.size(); i++)
  {
    if(bots[i].owner() == playerID())
    {
      bots[i].move(direction[rand()%4]);
    }
    if(bots[i].buildRate())
    {
      bots[i].build(types[rand() % types.size()], bots[i].x() + 1, bots[i].y(), 1);
    }
  }
  return true;
}

//This function is run once, after your last turn.
void AI::end(){}
